internal/config/main.go

```diff
@@ -125,12 +125,10 @@ func (c *Config) AutoDetect() error {
 	}
 
 	if changed {
-		for _, callback := range c.onChange {
-			if err := callback(); err != nil {
-				// we don't fail if the callback failed, as the auto-detection itself
-				// did work
-				c.logger.Error(err, "configuration change notification failed for callback")
-			}
+		if err := c.onChange.Change(); err != nil {
+			// we don't fail if the callback failed, as the auto-detection itself
```

Comment:
Again, nitpicking, maybe "Don't fail if the callback failed, as auto-detection itself worked"